src/Internal/RestockRefundedItemsAdjuster.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * RestockRefundedItemsAdjuster class file.
+ */
+
+namespace Automattic\WooCommerce\Internal;
+
+use Automattic\WooCommerce\Proxies\LegacyProxy;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class to adjust or initialize the restock refunded items.
+ */
+class RestockRefundedItemsAdjuster {
+	/**
+	 * The order factory to use.
+	 *
+	 * @var WC_Order_Factory
+	 */
+	private $order_factory;
+
+	/**
+	 * Class initialization, to be executed when the class is resolved by the container.
+	 *
+	 * @internal
+	 */
+	final public function init() {
+		$this->order_factory = wc_get_container()->get( LegacyProxy::class )->get_instance_of( \WC_Order_Factory::class );
+		add_action( 'woocommerce_before_save_order_items', array( $this, 'initialize_restock_refunded_items' ), 10, 2 );
+	}
+
+	/**
+	 * Initializes the restock refunded items meta for order version less than 5.4.
+	 *
+	 * @see https://github.com/woocommerce/woocommerce/issues/29502
+	 *
+	 * @param int   $order_id Order ID.
+	 * @param array $items Order items to save.
+	 */
+	public function initialize_restock_refunded_items( $order_id, $items ) {
+		$order         = wc_get_order( $order_id );
+		$order_version = $order->get_version();
+
+		if ( version_compare( $order_version, '5.4', '>=' ) ) {
+			return;
+		}
+
+		if ( isset( $items['order_item_id'] ) ) {
```

Comment:
perhaps we can bail early when there are no refunds to save processing time as well.

Reply:
Hmmm I was under the impression that we wanted to initialize this for all older orders that gets updated again? Otherwise we wouldn't need to do this here and could have just initialized it to zero if it doesn't exists which we saw it doesn't work as we know newer orders can have zero as legitimate value? Please elaborate more on this in case I am missing something.

Reply:
sorry, i meant that in the loop below, we are iterating over every line item and doing an insert. so I was just wondering, that if there are no refunds at all on the order that we are saving (whether stocked or not), then we can just bail early and save on that processing and make the request faster.

Reply:
Perhaps you would like to make a suggestion commit?